app/src/main/java/com/sthoray/allright/ui/main/view/HomeFragment.kt

```diff
@@ -41,6 +41,15 @@ class HomeFragment : Fragment(R.layout.fragment_home) {
         setupRecyclerView()
         setOnClickListeners()
         setupObservers()
+        swipeToRefreshHomeFragment()
+    }
+
+    private fun swipeToRefreshHomeFragment(){
+        swipeRefresh.setOnRefreshListener {
+            viewModel.getRefreshFragment()
+            //Toast.makeText(context, "Page refreshed!", Toast.LENGTH_SHORT).show()
+            swipeRefresh.isRefreshing = false
```

Comment:
This causes the swipe to refresh spinner is cancelled as soon as it is started. The spinner should stop only when data has successfully been fetched and is being displayed on the screen.

Reply:
The function on line 49 is called before the spinner is cancelled. The spinner will only cancel after we get the update data, if we don't set it to false at the end then the spinner will go forever.

Reply:
You function calls `getFeaturedCategories()`.  `getFeaturedCategories` launches a new coroutine and immediately finishes. This means your `getRefreshFragment` function does not block while network request are being made and cannot be used to signal when we are no longer refreshing. `isRefreshing` must be set back to false when we observe the response.